File: PASCAL/Cap07/c07ex08.pas
program c07ex08;

function FATORIAL(N: Integer): Integer;
begin
  if (N = 0) then
    FATORIAL := 1
  else
    FATORIAL := N * FATORIAL(N - 1);
end;

var
  LIMITE: Integer;

begin
  Write('Qual fatorial: ');
  Readln(LIMITE);

  Writeln('Fatorial = ', FATORIAL(LIMITE));

  Writeln;
  Write('Tecle <Enter> para encerrar... ');
  Readln;
end.

File: JAVASCRIPT/Cap07/c07ex08.js
// c07ex08.js

const rls = require('readline-sync');

function FATORIAL(N) {
  if (N == 0)
    return 1;
  else
    return N * FATORIAL(N - 1);
}

let Limite;

Limite = rls.questionInt("Qual fatorial: ");

console.log(`Fatorial = ${FATORIAL(Limite)}`);

console.log("");
process.stdout.write("Tecle <Enter> para encerrar... ");
rls.question();
